test/sleep-test.js

```diff
@@ -54,8 +60,9 @@ describe('Sleep Tests', () => {
       expect(sleep.getAllUsersAvgSleepQuality()).to.equal(3.5);
     });
 
-    it.skip('should be able to find the top best sleepers', () => {
-      // expect(sleep.findAllHighQualitySleepers()).to.equal()
+    it('should be able return users who average a sleep quality greater than 3 for a given week (7 days) - you should be able to calculate this for any week, not just the latest week', () => {
+      userRepo = new UserRepository(userData);
+      expect(sleep.findAllHighQualitySleepers(['2019/06/14', '2019/06/21']).to.deep.equal([2, 3, 5]));
```

Comment:
This code is working on front end. But having issues with the test suite.

Reply:
Pulled down this branch. The test is failing because you are trying to use userRepo inside your sleep class on line 88. Sleep class doesn't know about userRepo so that's why it's giving an error. I think it works in the front end bc it is global in the application so sleep can pull that info in the front end.

I think the test would pass if you create a new instance of user repo in the sleep class.

Reply:
I tried doing that, and I tried creating a userRepo in the test as well. I could not seem to get it working. I'm going to keep playing with it.